### PR TITLE
Use hypothesis.assume to avoid close quantiles

### PR DESCRIPTION
**Pull request by TTitscher**
In the fuzzy testing, we basically reformulate the previously made requirements (like x0 > x1) properly using [`hypothesis.assume`](https://hypothesis.readthedocs.io/en/latest/details.html?highlight=assume#hypothesis.assume).

### CODE REVIEW

File: tests/test_gamma.py [hunk 1]
```diff
@@ -26,10 +26,8 @@ def test_sd():
 )
 def test_from_quantiles(x0_x1):
     x0, x1 = x0_x1
-    if x0 == x1:
-        return
-    if x0 > x1:
-        x1, x0 = x0, x1
+    assume(abs(x1 - x0) > 1e-6)
+    assume(x0 < x1)
```
Comment (ajafarihub) [inline tests/test_gamma.py:30]: Since we ordered x0 and x1 just above, this might be unnecessary.

File: tests/test_gamma.py [hunk 2]
```diff
@@ -26,10 +26,8 @@ def test_sd():
...
 
     q = (0.15, 0.95)
```
Comment (ajafarihub) [inline tests/test_gamma.py:32]: Wouldn't (0.05, 0.95) be more natural? Or is there any reason for having 0.15 ?

File: tests/test_gamma.py [hunk 3]
```diff
@@ -26,10 +26,8 @@ def test_sd():
 )
 def test_from_quantiles(x0_x1):
     x0, x1 = x0_x1
-    if x0 == x1:
-        return
-    if x0 > x1:
-        x1, x0 = x0, x1
+    assume(abs(x1 - x0) > 1e-6)
```
Comment (ajafarihub) [inline tests/test_gamma.py:29]: I wondered why 1e-6? Would that limit concern with the tolerance that - I suppose - is used [here](https://github.com/BAMresearch/bayem/blob/1fd1800127b16f5cbb37110acc92012c2ff3a96e/bayem/distributions.py#L187)?

File: tests/test_gamma.py (hunk 1 above)
Comment (TTitscher) [reply tests/test_gamma.py:30]: You mean the abs?

File: tests/test_gamma.py (hunk 2 above)
Comment (TTitscher) [reply tests/test_gamma.py:32]: Yes typo, I guess. I didn't touch that in this PR, but go ahead and change it, if you want.

File: tests/test_gamma.py (hunk 3 above)
Comment (TTitscher) [reply tests/test_gamma.py:29]: The issue for the previously failing test was x0 and x1 being too close. I guess smaller tolerances would also work, but the Gamma distributions with such close quantiles are never used, I assume. Let me know, if I should add this explanation as a comment.

File: tests/test_gamma.py (hunk 1 above)
Comment (ajafarihub) [reply tests/test_gamma.py:30]: Sorry, it was my mistake. That is fine.

File: tests/test_gamma.py (hunk 3 above)
Comment (ajafarihub) [reply tests/test_gamma.py:29]: I guess, some comments could be useful.

File: tests/test_gamma.py (hunk 2 above)
Comment (ajafarihub) [reply tests/test_gamma.py:32]: Maybe it is not too important, then.

**Comment by ajafarihub** [comment]
The pull request is OK from my point of view.

**Pull request merged**